editor/layout/index.js

```diff
@@ -11,26 +11,55 @@ import Header from '../header';
 import Sidebar from '../sidebar';
 import TextEditor from '../modes/text-editor';
 import VisualEditor from '../modes/visual-editor';
-import { getEditorMode, isEditorSidebarOpened } from '../selectors';
-
-function Layout( { mode, isSidebarOpened } ) {
-	const className = classnames( 'editor-layout', {
-		'is-sidebar-opened': isSidebarOpened,
-	} );
-
-	return (
-		<div className={ className }>
-			<Header />
-			<div className="editor-layout__content">
-				{ mode === 'text' && <TextEditor /> }
-				{ mode === 'visual' && <VisualEditor /> }
+import {
+	getEditorMode,
+	isEditorSidebarOpened,
+	getEditorWidth,
+} from '../selectors';
+
+class Layout extends wp.element.Component {
+
+	componentDidMount() {
+		window.addEventListener( 'resize', this.props.setEditorWidth );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'resize', this.props.setEditorWidth );
```

Comment:
We should confirm that `mapDispatchToProps` is only called once for the lifetime of the component. Otherwise we run the risk of the reference here to `setEditorWidth` not being the same as the one assigned in `componentDidMount`, leaving zombie event listeners.